src/appleseed/renderer/kernel/lighting/lighttree.cpp

```diff
@@ -0,0 +1,332 @@
+                                       + ".py";
+            foundation::VPythonFile file(filename.c_str());
+            file.draw_axes(Width);
+
+            // Draw the initial bbox.
+            file.draw_aabb(root_bbox, color, Width);
+
+            // Find the parent node to draw child bboxes from.
+            for (size_t i = 0; i < m_nodes.size(); i++)
+            {
+                if (m_nodes[i].is_leaf())
+                    continue;
+
+                if (m_nodes[i].get_level() == parent_level)
+                {
+
+                    const auto& bbox_left = m_nodes[i].get_left_bbox();
+                    const auto& bbox_right = m_nodes[i].get_right_bbox();
+
+                    file.draw_aabb(bbox_left, color, Width);
+                    file.draw_aabb(bbox_right, color, Width);
+                }
+            }
+        }
+    }
+    else
+    {
+        const std::string filename = filename_base + ".py";
+        foundation::VPythonFile file(filename.c_str());
+        file.draw_axes(Width);
+        const char* color = "color.yellow";
+
+        // Draw the initial bbox.
+        file.draw_aabb(root_bbox, color, Width);
+
+        // Find the parent node to draw child bboxes from.
+        for (size_t i = 0; i < m_nodes.size(); i++)
+        {
+            if (m_nodes[i].is_leaf())
+                continue;
+
+            // Make even levels red and odd green
+            color = m_nodes[i].get_level() % 2 ? "color.red"
+                                               : "color.green";
+
+            const auto& bbox_left = m_nodes[i].get_left_bbox();
+            const auto& bbox_right = m_nodes[i].get_right_bbox();
+
+            file.draw_aabb(bbox_left, color, Width);
+            file.draw_aabb(bbox_right, color, Width);
+        }
+    }
+}
+
+//
+// Calculate the tree depth and assign total luminance to each node of the tree.
+// Total luminance represents the sum of all its child nodes luminances.
+//
+
+float LightTree::recursive_node_update(size_t node_index, size_t node_level)
+{
+    float luminance = 0.0f;
+
+    if (!m_nodes[node_index].is_leaf())
+    {
+        const auto& child1 = m_nodes[node_index].get_child_node_index();
+        const auto& child2 = m_nodes[node_index].get_child_node_index() + 1;
+
+        float luminance1 = recursive_node_update(child1, node_level + 1);
+        float luminance2 = recursive_node_update(child2, node_level + 1);
+
+        luminance = luminance1 + luminance2;
+    }
+    else
+    {
+        // Access the light intensity value.
+        size_t item_index = m_nodes[node_index].get_item_index();
+        size_t light_source_index = m_items[item_index].m_light_index;
+        Spectrum spectrum = m_light_sources[light_source_index]->get_intensity();
+
+        // Luminance is the average accross all the spectrum channels.
+        for (size_t i = 0; i < spectrum.size(); i++)
+        {
+            luminance += spectrum[i];
+        }
+        luminance /= spectrum.size();
```

Comment:
I believe this won't compile on Windows (where we have more aggressive warnings enabled) since there's an implicit (and potentially lossy) conversion from `size_t` to `float`. You need to add a cast:

    luminance /= static_cast<float>(spectrum.size());

But now that I look at it properly, all you need to do is

    luminance = foundation::average_value(spectrum);